Populate table import maps

viniciusdc opened the pull request.

viniciusdc committed:
mv transverse function to utils & add back path to FeedstockOutputs

viniciusdc committed:
Add initial populate function for import to package maps

viniciusdc committed:
add populate method for import to package mappings

viniciusdc committed:
Add action to automate the DB testing and replication

viniciusdc committed:
Extend CI condition triggers

viniciusdc committed:
add if always condtional to workflow run

viniciusdc committed:
fix pathing

viniciusdc committed:
Add setup.py for easy call during CI

viniciusdc committed:
Pip install cfdb

viniciusdc committed:
Fix directory paths

viniciusdc committed:
Fix wrong pathing

viniciusdc committed:
add missing dir if always

viniciusdc committed:
rm use of popd to fix stack trace error and add json blob index count - find

viniciusdc committed:
list files in root dir

viniciusdc committed:
change to download/upload artifacts

viniciusdc committed:
update artifact download path

jaimergp commented on .github/workflows/build_db.yaml:
```diff
@@ -0,0 +1,75 @@
+      uses: actions/download-artifact@v2
+      with:
+        name: cfdb
+        path: /home/runner/work/czi-conda-forge-db/cf-database.db
```
```suggestion
        path: ${{ github.workspace }}/../cf-database.db
```

This is equivalent but why on the directory above?

viniciusdc replied:
Thanks, this could be changed if needed, though, then we would need to pass the path to the new location into the execution command as, by default, it will look into the dir that called the command for the DB file

jaimergp commented on .github/workflows/build_db.yaml:
```diff
@@ -0,0 +1,75 @@
+        path: /home/runner/work/czi-conda-forge-db/cf-database.db
+
+    - name: Create runtest directory
+      if: always()
```
Why do you need all these `always()` calls?

viniciusdc replied:
Because of the order of which everything is done right now, to use the artifacts as a form of caching the previous state of the DB in the workflow above, we need first to download it from the artifacts store. The issue with the very first run is that such object does not exists, and the artifact action above will return an error and fail when  that happens, which thus hinder the progress of the pipeline that would have created the file and pushed it

jaimergp commented on .github/workflows/build_db.yaml:
```diff
@@ -0,0 +1,75 @@
+        mkdir database
+        cd database
+
+    - name: Clone Feedstock Outputs
+      if: always()
+      run: |
+        git clone https://github.com/conda-forge/feedstock-outputs.git
+        echo "FEEDSTOCK_OUTPUTS=$(pwd)/feedstock-outputs" >> $GITHUB_ENV
```
You can use `actions/checkout` for this too.

viniciusdc replied:
That I didn't know, thanks. This can be added at a later time then (I will need to revisit those once we switch the source)

jaimergp commented on .github/workflows/build_db.yaml:
```diff
@@ -0,0 +1,75 @@
+      run: |
+        mkdir libcfgraph && cd libcfgraph
+        git init
+        git config core.sparsecheckout true
```
Nice trick! Could be useful in other cf repos.

jaimergp commented on cfdb/main.py:
```diff
@@ -86,6 +97,24 @@ def update_feedstock_outputs(
+    )
+):
+    """
+    Update the import to package maps in the database based on the local path to the import to package maps cloned from Conda Forge. Path to the import to package maps directory. The path should point to the 'import_to_package_maps' folder inside the 'libcfgraph' root directory or any viable alternative.
```
```suggestion
    Update the import to package maps in the database based on the local path to the
    import to package maps cloned from conda-forge. Path to the import to package
    maps directory. The path should point to the 'import_to_package_maps' folder
    inside the 'libcfgraph' root directory or any viable alternative.
```

jaimergp commented on cfdb/main.py:
```diff
@@ -86,6 +97,24 @@ def update_feedstock_outputs(
+    """
+    Update the import to package maps in the database based on the local path to the import to package maps cloned from Conda Forge. Path to the import to package maps directory. The path should point to the 'import_to_package_maps' folder inside the 'libcfgraph' root directory or any viable alternative.
+
+
```
```suggestion
```

jaimergp commented on cfdb/populate/import_to_package_maps.py:
```diff
@@ -0,0 +1,112 @@
+
+    Args:
+        session (Session): The SQLAlchemy session object.
+        path (Path): The path to import to package maps directory containing the JSON blobs. (relative to the root directory of "libcfgraph" or viable alternative).
```
```suggestion
        path (Path): The path to import to package maps directory containing the JSON blobs
        (relative to the root directory of "libcfgraph" or viable alternative).
```

jaimergp commented on cfdb/populate/utils.py:
```diff
@@ -55,3 +59,77 @@ def retrieve_associated_feedstock_from_output_blob(file: Path):
+    return packages_to_imports
+
+
+def transverse_files(path: Path, output_dir: Path = None) -> List[Path]:
```
I think you mean `traverse_*`

viniciusdc replied:
Thanks

jaimergp commented on setup.py:
Let's use `pyproject.toml` for this.

viniciusdc replied:
Done, thanks

jaimergp commented on tests/unitests/test_schema.py:
```diff
@@ -23,6 +23,7 @@ def sample_package() -> Packages:
 def sample_feedstock_output(sample_feedstock, sample_package) -> FeedstockOutputs:
     return FeedstockOutputs(
         id="output_1",
+        path="/path/to/output_1",
```
Hopefully we can uncouple this in the future. I do believe it shouldn't be part of the database tables. After all it's just an implementation detail in the original repo-database.

viniciusdc replied:
Indeed, inspecting the `libcflib` repo I found a new way to source data, this will be removed in the next PR

jaimergp commented on cfdb/populate/import_to_package_maps.py:
```diff
@@ -0,0 +1,112 @@
+
+                for _import in imports:
+                    _mapping = ImportToPackageMaps(
+                        id=uniq_id(),  # to update
```
"To update" what exactly?

viniciusdc replied:
good question :thinking: , I will remove it as IDK what I meant (I guess I was using int as the `unique` identifier for that table before, and switched to that function with a global replace, and missed the comment)

jaimergp reviewed (commented):
A few comments that need to be addressed but nothing that would require a major change in our direction. I am only concerned about the JSON path detail leaking into our DB schema, perhaps unnecessarily.

viniciusdc committed:
Update download-artifact file path

Suggested during review by Jaime

Co-authored-by: jaimergp <jaimergp@users.noreply.github.com>

viniciusdc committed:
Update cfdb/main.py

Co-authored-by: jaimergp <jaimergp@users.noreply.github.com>

viniciusdc committed:
Update cfdb/main.py

Co-authored-by: jaimergp <jaimergp@users.noreply.github.com>

viniciusdc committed:
Update cfdb/populate/import_to_package_maps.py

Co-authored-by: jaimergp <jaimergp@users.noreply.github.com>

viniciusdc committed:
Fix misspelling

viniciusdc committed:
Move to pyproject.toml

viniciusdc committed:
rm unecessary comment line

viniciusdc committed:
Fix issue with author/authors section on toml file

viniciusdc committed:
Fix issue with author/authors section on toml file

viniciusdc committed:
Fix issue with author/authors section on toml file

viniciusdc committed:
Update to more concise worklfow configuration

viniciusdc committed:
Update to more concise worklfow configuration

Update to more concise worklfow configuration

Update to more concise worklfow configuration

Update to more concise worklfow configuration

viniciusdc committed:
Test artifact DB import
